libs/remote_storage/src/s3_bucket.rs

```diff
@@ -83,10 +87,23 @@ impl S3Bucket {
             .or_else("imds", ImdsCredentialsProvider::builder().build())
         };
 
+        // AWS SDK requires us to specify how the RetryConfig should sleep when it wants to back off
+        let sleep_impl: Arc<dyn AsyncSleep> = Arc::new(TokioSleep::new());
+
+        // We do our own retries (see [`backoff::retry`]).  However, for the AWS SDK to enable rate limiting in response to throttling
+        // responses (e.g. 429 on too many ListObjectsv2 requests), we must provide a retry config.  We set it to use at most one
+        // attempt, and enable 'Adaptive' mode, which causes rate limiting to be enabled.
```

Comment:
some points on this:

* [the default is 3](https://docs.rs/aws-smithy-types/0.57.1/src/aws_smithy_types/retry.rs.html#270)
* the first try [also counts into the number of attempts](https://docs.rs/aws-config/latest/aws_config/retry/struct.RetryConfig.html#method.with_max_attempts)